java/src/main/java/com/anaconda/skein/ApplicationMaster.java

```diff
@@ -576,8 +576,13 @@ public synchronized void initialize() throws IOException {
 
     private synchronized void requestContainer(Model.Container container) {
       Priority priority = newPriority(this);
-      ContainerRequest req = new ContainerRequest(service.getResources(),
-                                                  null, null, priority);
+      ContainerRequest req = new ContainerRequest(
+          service.getResources(),
+          null,
+          null,
+          priority,
+          true,
+          service.getNodeLabel().isEmpty() ? null : service.getNodeLabel());
```

Comment:
IIUC this should also potentially inherit from the application wide node label. I'd probably propogate the application wide default down the services in `loadApplicationSpec` or in the `ServiceTracker` constructor.

Reply:
Makes sense, will do.